pkg/terra/client/test_helpers.go

```diff
@@ -75,49 +79,73 @@ func SetupLocalTerraNode(t *testing.T, chainID string) ([]Account, string) {
 		privateKey, address := createKeyFromMnemonic(t, k.Mnemonic)
 		require.Equal(t, expAcctAddr, address)
 		// Give it 100 luna
-		_, err = exec.Command("terrad", "add-genesis-account", k.Address, "100000000uluna", "--home", testdir).Output()
-		require.NoError(t, err)
+		out2, err2 := exec.Command("terrad", "add-genesis-account", k.Address, "100000000uluna", "--home", testdir).Output() //nolint:gosec
+		require.NoError(t, err2, string(out2))
 		accounts = append(accounts, Account{
 			Name:       account,
 			Address:    address,
 			PrivateKey: privateKey,
 		})
 	}
 	// Stake 10 luna in first acct
-	out, err := exec.Command("terrad", "gentx", accounts[0].Name, "10000000uluna", fmt.Sprintf("--chain-id=%s", chainID), "--keyring-backend", "test", "--keyring-dir", testdir, "--home", testdir).CombinedOutput()
+	out, err = exec.Command("terrad", "gentx", accounts[0].Name, "10000000uluna", fmt.Sprintf("--chain-id=%s", chainID), "--keyring-backend", "test", "--keyring-dir", testdir, "--home", testdir).CombinedOutput() //nolint:gosec
 	require.NoError(t, err, string(out))
 	out, err = exec.Command("terrad", "collect-gentxs", "--home", testdir).CombinedOutput()
 	require.NoError(t, err, string(out))
-	cmd := exec.Command("terrad", "start", "--home", testdir)
+
+	port := mustRandomPort()
+	tendermintURL := fmt.Sprintf("http://127.0.0.1:%d", port)
+	t.Log(tendermintURL)
+	cmd := exec.Command("terrad", "start", "--home", testdir,
+		"--rpc.laddr", fmt.Sprintf("tcp://127.0.0.1:%d", port),
+		"--rpc.pprof_laddr", "0.0.0.0:0",
+		"--grpc.address", "0.0.0.0:0",
+		"--grpc-web.address", "0.0.0.0:0",
+		"--p2p.laddr", "0.0.0.0:0")
+	var stdErr bytes.Buffer
+	cmd.Stderr = &stdErr
 	require.NoError(t, cmd.Start())
 	t.Cleanup(func() {
-		require.NoError(t, cmd.Process.Kill())
+		assert.NoError(t, cmd.Process.Kill())
+		if err2 := cmd.Wait(); assert.Error(t, err2) {
+			if !assert.Contains(t, err2.Error(), "signal: killed", cmd.ProcessState.String()) {
+				t.Log("terrad stderr:", stdErr.String())
+			}
+		}
```

Comment:
This was the key to diagnosing the core CI failures.

Reply:
curious what the error string was?

Reply:
It would immediately exiting complaining about one of the ports, and dump a command help menu, but that didn't trigger a normal error.